### PR TITLE
[PROF-9744] Propagate errors from expect calls

### PR DESCRIPTION
**Pull request by taegyunkim**
# What does this PR do?

Propagates errors from unwrap/expect calls instead of panicking right away. 

# Motivation

These might help reducing crashes from our profilers. 

# Additional Notes

Anything else we should know when reviewing?

# How to test the change?

`cargo test`

### CODE REVIEW

File: profiling/src/internal/profile.rs [hunk 1]
```diff
@@ -370,10 +370,13 @@ impl Profile {
     }
 
     fn get_endpoint_for_labels(&self, label_set_id: LabelSetId) -> anyhow::Result<Option<Label>> {
-        let label = self.get_label_set(label_set_id).iter().find_map(|id| {
+        let label = self.get_label_set(label_set_id)?.iter().find_map(|id| {
```
Comment (danielsn) [inline profiling/src/internal/profile.rs:373]: do `if let Ok(label)` ...

File: profiling/src/internal/profile.rs [hunk 2]
```diff
@@ -385,22 +388,22 @@ impl Profile {
...
-            .expect("LabelSetId to have a valid interned index")
+            .ok_or(anyhow::anyhow!("LabelSetId to have a valid interned index"))
     }
 
-    fn get_stacktrace(&self, st: StackTraceId) -> &StackTrace {
+    fn get_stacktrace(&self, st: StackTraceId) -> anyhow::Result<&StackTrace> {
         self.stack_traces
             .get_index(st.to_raw_id())
-            .expect("StackTraceId {st} to exist in profile")
+            .ok_or(anyhow::anyhow!("StackTraceId {:?} to exist in profile", st))
```
Comment (danielsn) [inline profiling/src/internal/profile.rs:406]: `with_context` is another option

File: profiling/src/internal/profile.rs [hunk 3]
```diff
@@ -385,22 +388,22 @@ impl Profile {
...
-            .expect("LabelId to have a valid interned index")
+            .ok_or(anyhow::anyhow!("LabelId to have a valid interned index"))
     }
 
-    fn get_label_set(&self, id: LabelSetId) -> &LabelSet {
+    fn get_label_set(&self, id: LabelSetId) -> anyhow::Result<&LabelSet> {
         self.label_sets
             .get_index(id.to_offset())
-            .expect("LabelSetId to have a valid interned index")
+            .ok_or(anyhow::anyhow!("LabelSetId to have a valid interned index"))
```
Comment (danielsn) [inline profiling/src/internal/profile.rs:400]: `.context` is another option

File: profiling/src/internal/upscaling.rs [hunk 4]
```diff
@@ -85,7 +85,7 @@ impl UpscalingRules {
                 let (_, rules) = self
                     .rules
                     .get_index_mut(index)
-                    .expect("Already existing rules");
+                    .ok_or(anyhow::anyhow!("Already existing rules"))?;
```
Comment (morrisonlevi) [inline profiling/src/internal/upscaling.rs]: I think we could improve on the messaging here while we are at it. Just an idea:
```suggestion
                    .ok_or(anyhow::anyhow!("Expected upscaling rule to exist for index {index}"))?;
```

File: profiling/src/internal/upscaling.rs [hunk 5]
```diff
@@ -85,7 +86,7 @@ impl UpscalingRules {
                 let (_, rules) = self
                     .rules
                     .get_index_mut(index)
-                    .expect("Already existing rules");
+                    .context("Already existing rules")?;
```
Comment (morrisonlevi) [inline profiling/src/internal/upscaling.rs:89]: Maybe we can improve the message at the same time? Something like:

```suggestion
                    .context("Expected upscaling rule to exist for index {index}")?;
```

**Review by danielsn — APPROVED** [review]

File: profiling/src/internal/upscaling.rs [hunk 6]
```diff
@@ -82,10 +83,9 @@ impl UpscalingRules {
                 self.rules.insert((label_name_id, label_value_id), rules);
             }
             Some(index) => {
-                let (_, rules) = self
-                    .rules
-                    .get_index_mut(index)
-                    .expect("Already existing rules");
+                let (_, rules) = self.rules.get_index_mut(index).with_context(|| {
+                    format!("Expected upscaling rules to exist for index {}", index)
```
Comment (morrisonlevi) [inline profiling/src/internal/upscaling.rs:87]: Does this work?

```suggestion
                    format!("Expected upscaling rules to exist for index {index}")
```

**Comment by ivoanjo** [comment]
No merge queue on this repo ;)

**Pull request merged**